test/functional/mempool_wallet_transactions.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# Copyright (c) 2009-2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Ensure that wallet transactions get successfully broadcast to at least one peer.
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+        assert_greater_than,
+        wait_until,
+        create_confirmed_utxos,
+        disconnect_nodes
+)
+from test_framework.mininode import P2PTxInvStore
+import time
+
+# Constant from txmempool.h
+MAX_REBROADCAST_WEIGHT = 3000000
+
+class MempoolWalletTransactionsTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 2
+        self.extra_args = [[
+            "-whitelist=127.0.0.1",
+            "-blockmaxweight=3000000"
+            ]] * self.num_nodes
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        self.log.info("test that mempool will ensure initial broadcast of wallet txns")
+
+        node = self.nodes[0]
+        min_relay_fee = node.getnetworkinfo()["relayfee"]
+        utxos = create_confirmed_utxos(min_relay_fee, node, 2000)
+
+        self.log.info("populate mempool with transactions")
+        node.settxfee(min_relay_fee * 3)
+
+        addresses = []
+        for _ in range(50):
+            addresses.append(self.nodes[1].getnewaddress())
+
+        # create large txns by sending to all the addresses
+        outputs = {}
+        for addr in addresses:
+            outputs[addr] = 0.0001
+
+        self.sync_mempools()
+
+        # create lots of txns with that large output
+        for _ in range(len(utxos) - 1):
+            utxo = utxos.pop()
+            inputs = [{'txid': utxo['txid'], 'vout': utxo['vout']}]
+            raw_tx_hex = node.createrawtransaction(inputs, outputs)
+            signed_tx = node.signrawtransactionwithwallet(raw_tx_hex)
+            node.sendrawtransaction(hexstring=signed_tx['hex'], maxfeerate=0)
+            self.nodes[1].sendrawtransaction(hexstring=signed_tx['hex'], maxfeerate=0)
```

Comment:
You are creating blocks on node[0], that this node (node[1]) might not have. This will lead to a race.

You can avoid it by calling sync_blocks or sync_all after you have generated the blocks.

Reply:
ah, got it now. thank you!